### ISSUE TITLE
Add `bbox` parameter to @turf/interpolate

### ISSUE THREAD

**Issue by stebogit** [Turfjs/turf#1031]
Currently `@turf/interpolate` defines the area on which to interpolate the values as the bounding box surrounding the input points.
It would be useful to be able to define your area of interest, which might be different than the current (then default)`bbox(points)`, like this:

![screen shot 2017-10-19 at 8 11 05 am](https://user-images.githubusercontent.com/12717225/31778324-33471938-b4a5-11e7-9a09-755edd7aae3f.png)

or this:
![screen shot 2017-10-19 at 8 19 22 am](https://user-images.githubusercontent.com/12717225/31778729-42ec3d72-b4a6-11e7-984e-f64ce4afb550.png)

**Comment by DenisCarriere** [comment]
We can pass the same `options` from the grid modules, I've added `mask` for each of them, that might work?

**Comment by stebogit** [comment]
Yep, `mask` would definitely work 👍 

Now that I look at it we should also rename `cellSize` to `cellSide` as we did in the grid modules

Also, I was wondering why was `clone` introduced in [here](https://github.com/Turfjs/turf/commit/6e97e3043f5b363d487eed626373d3b28425bc2e#diff-cd650bfe6ae7b9c5243511adf496138fR90)
```js
var newFeature = clone(gridFeature);
newFeature.properties[property] = zw / sw;
results.push(newFeature);
...
return featureCollection(results);
```

**Comment by DenisCarriere** [comment]
Clone was used because it mutates the properties, I'm sure there might be a way to remove the clone.


```js
newFeature.properties[property] = zw / sw;
```

**Comment by DenisCarriere** [comment]
- Change to `cellSide` 👍 
- `mask` works 👍

**Comment by romick101** [comment]
Hello, is there any progress with `mask` parameter? Maybe I'm overlooking some other way to achieve custom interpolation boundary?

**Comment by iamtekson** [comment]
Is the `mask` parameter already implemented? I haven't seen that in parameter.

**Comment by smallsaucepan** [comment]
Hi @iamtekson. Could you please clarify if this is currently just a documentation issue ...

> mask parameter [...] is there and working properly

or an actual problem with the result you get back from the function ...

> but I got the values only for the bbox of the input points not for the whole polygon

Or both?

**Comment by smallsaucepan** [comment]
This will be resolved as part of #2444 so the user can pass an optional bbox.
